src/sentences/aam.rs

```diff
@@ -0,0 +1,177 @@
+use arrayvec::ArrayString;
+use nom::{
+    bytes::complete::is_not,
+    character::complete::{anychar, char},
+    combinator::opt,
+    number::complete::float,
+};
+
+use crate::{parse::NmeaSentence, sentences::utils::array_string, Error, SentenceType};
+
+const MAX_LEN: usize = 64;
+
+/// AAM - Waypoint Arrival Alarm
+///
+/// <https://gpsd.gitlab.io/gpsd/NMEA.html#_aam_waypoint_arrival_alarm>
+///
+/// ```text
+///        1 2 3   4 5    6
+///        | | |   | |    |
+/// $--AAM,A,A,x.x,N,c--c*hh<CR><LF>
+///
+/// Field Number:
+///   1. Status, BOOLEAN, A = Arrival circle entered, V = not passed
+///   2. Status, BOOLEAN, A = perpendicular passed at waypoint, V = not passed
+///   3. Arrival circle radius
+///   4. Units of radiuos, nautical miles
+///   5. Waypoint ID
+///   6. Checksum
+///
+/// Example: $GPAAM,A,A,0.10,N,WPTNME*43
+/// WPTNME is the waypoint name.
+/// ```
+#[derive(Debug, PartialEq)]
+pub struct AamData {
+    pub arrival_circle_entered: Option<bool>,
+    pub perpendicular_passed: Option<bool>,
+    pub arrival_circle_radius: Option<f32>,
+    pub radius_units: Option<char>,
+    pub waypoint_id: Option<ArrayString<MAX_LEN>>,
+}
+
+/// Parse AAM message
+pub fn parse_aam(sentence: NmeaSentence) -> Result<AamData, Error> {
+    if sentence.message_id != SentenceType::AAM {
+        Err(Error::WrongSentenceHeader {
+            expected: SentenceType::AAM,
+            found: sentence.message_id,
+        })
+    } else {
+        Ok(do_parse_aam(sentence.data)?)
+    }
+}
+
+fn do_parse_aam(i: &str) -> Result<AamData, Error> {
+    let (i, arrival_circle_entered) = anychar(i)?;
+    let arrival_circle_entered = match arrival_circle_entered {
+        'A' => Some(true),
+        'V' => Some(false),
+        char => unreachable!("{} is not a valid AAM-Arrival Circle Entered value", char),
+    };
+    let (i, _) = char(',')(i)?;
+
+    let (i, perpendicular_passed) = anychar(i)?;
+    let perpendicular_passed = match perpendicular_passed {
+        'A' => Some(true),
+        'V' => Some(false),
+        char => unreachable!("{} is not a valid AAM-Perpendicular Passed value", char),
```

Comment:
only when using `one_of` this will be unreachable.
Right now, because of `anychar` it will match other values like `G` and will reach the `unreachable!` call.